windows/CodePushPackage.cs

```diff
@@ -0,0 +1,364 @@
+            return await codePushFolder.CreateFileAsync(STATUS_FILE, CreationCollisionOption.OpenIfExists);
+        }
+
+        public async Task<JObject> GetCurrentPackageInfo()
+        {
+            StorageFile statusFile = await GetStatusFile();
+            try
+            {
+                return await CodePushUtils.GetJObjectFromFile(statusFile);
+            }
+            catch (Exception e)
+            {
+                throw new CodePushUnknownException("Error getting current package info", e);
+            }
+        }
+
+        public async Task UpdateCurrentPackageInfo(JObject packageInfo)
+        {
+            try
+            {
+                await FileIO.WriteTextAsync(await GetStatusFile(), JsonConvert.SerializeObject(packageInfo));
+            }
+            catch (IOException e)
+            {
+                throw new CodePushUnknownException("Error updating current package info", e);
+            }
+        }
+
+        public async Task<StorageFolder> GetCurrentPackageFolder()
+        {
+            JObject info = await GetCurrentPackageInfo();
+            string packageHash = (string)info[CURRENT_PACKAGE_KEY];
+            if (packageHash == null)
+            {
+                return null;
+            }
+
+            return await GetPackageFolder(packageHash, false);
+        }
+
+
+        public async Task<StorageFile> GetCurrentPackageBundle(string bundleFileName)
+        {
+            StorageFolder packageFolder = await GetCurrentPackageFolder();
+            if (packageFolder == null)
+            {
+                return null;
+            }
+
+            JObject currentPackage = await GetCurrentPackage();
+            string relativeBundlePath = (string)currentPackage[RELATIVE_BUNDLE_PATH_KEY];
+            if (relativeBundlePath == null)
+            {
+                return await packageFolder.GetFileAsync(bundleFileName);
+            }
+            else
+            {
+                return await packageFolder.GetFileAsync(relativeBundlePath);
+            }
+        }
+
+        public async Task<StorageFolder> GetPackageFolder(string packageHash, bool createIfNotExists)
+        {
+            StorageFolder codePushFolder = await GetCodePushFolder();
+            try
+            {
+                packageHash = shortenPackageHash(packageHash);
+                if (createIfNotExists)
+                {
+                    return await codePushFolder.CreateFolderAsync(packageHash, CreationCollisionOption.OpenIfExists);
+                }
+                else
+                {
+                    return await codePushFolder.GetFolderAsync(packageHash);
+                }
+            }
+            catch (FileNotFoundException)
+            {
+                return null;
+            }
+        }
+
+        public async Task<string> GetCurrentPackageHash()
+        {
+            JObject info = await GetCurrentPackageInfo();
```

Comment:
GetCurrentPackageHash and GetPreviousPackageHash are nearly identical except using a different key...helper function?